### ISSUE TITLE
BUG:torch.nn.functional.multi_margin_loss

### ISSUE THREAD

**Issue by laizewei** [pytorch/pytorch#15166]
pytorch1.0
BUG:torch.nn.functional.multi_margin_loss
example:
import torch 
import torch.nn.functional as F
a=[[1.,2.,3.,4.]]
b=[2]
input=torch.tensor(a,requires_grad=True)
target = torch.tensor(b)
k = torch.tensor(c)
output = F.multi_margin_loss(input,target,p=1.,margin=1.,reduce=False,size_average=False)
print(output)
--------------------------
tensor([0.5000], grad_fn=<MultiMarginLossBackward>)

**Comment by albanD** [comment]
Hi,

Could you please elaborate a bit more on what is the problem here?

**Comment by albanD** [comment]
Hi,

The doc is correct (though a bit misleading), it states, just before the formula: `For each mini-batch sample, the loss in terms of the 1D input x and scalar output y is`. The loss(x, y) formula is given for a single element of the batch where x is a 1D tensor containing the scores and y is the value of the true label.
Notice that the formula sum over all possible labels different than the ground truth, so you get a single number for each sample.

Then the reductions specify how you want to aggregate results for the elements in the mini-batch. In your case, you have a single element in your minibatch, so reduction will always return the same thing.
